pkg/persistence/object_store_test.go

```diff
@@ -612,59 +615,75 @@ func TestNewObjectBackupStoreGetter(t *testing.T) {
 		name              string
 		location          *velerov1api.BackupStorageLocation
 		objectStoreGetter objectStoreGetter
+		credFileStore     credentials.FileStore
+		fileStoreErr      error
 		wantBucket        string
 		wantPrefix        string
 		wantErr           string
 	}{
 		{
-			name:     "location with no ObjectStorage field results in an error",
-			location: new(velerov1api.BackupStorageLocation),
-			wantErr:  "backup storage location does not use object storage",
+			name:          "location with no ObjectStorage field results in an error",
```

Comment:
Please follow the existing convention in this test block of "when/then" or. Also, no need to specify that what you are checking for is a field. I would rename this test...

from:
"location with no ObjectStorage field results in an error "

to:
"when location  doesn't use an object storage, a backup store can't be retrieved"

Notes on this:
- I'm helping myself to the existing error messages for this specific failure to make writing this easier
- Note I didn't mention it returns an error: the test case itself is showing me that it does
- Stating that when there isn't an object storage the consequence is that a backup store can't be retrieved is precise, whereas "returns an error" is not very informative. Also, an error in this method can be returned for many different reasons, so it's useful to sort out which is which.
- Think of writing test cases as writing a spec and not knowing what the code looks or will look like.

Note this test case name:
"when Prefix has a leading and trailing slash, the leading slash is stripped and the trailing slash is left" - it is much more revealing and helpful than if it had said: "succeeds".

Reply:
Those were the original test names, but I opted to follow that style instead of the spec style in the test that I added :woman_facepalming: 
You're right, the spec style is much more descriptive :+1: I'll update these tests and the ones I'm adding.